.github/workflows/build_and_test.yml

```diff
@@ -220,6 +220,11 @@ jobs:
           cd docker
           docker compose -f docker-compose.yml -f ${{ matrix.demo_name }}.yml up ${{ matrix.demo_name }} --no-build --exit-code-from ${{ matrix.demo_name }}
 
+      - name: Stop docker containers
+        run: |
+          cd docker
+          docker compose -f docker-compose.yml -f ${{ matrix.demo_name }}.yml stop
```

Comment:
I noticed that the docker containers kept on running, because I changed the way we use `docker compose .. up`. I target a specific service only, and rely on `depends_on` for all the dependent services to initialize. As a side effect though, it also meant that when that target service exited, its dependent services kept running. So I added a specific `docker compose stop` step here.